internal/js/modules/k6/browser/common/page.go

```diff
@@ -1418,6 +1418,44 @@ func (p *Page) Route(
 	return nil
 }
 
+// Unroute removes the route(s) for the specified URL pattern.
+// If multiple routes match the same URL pattern, all of them are removed.
+func (p *Page) Unroute(path string) error {
+	p.logger.Debugf("Page:Unroute", "sid:%v path:%s", p.sessionID(), path)
+
+	p.routesMu.Lock()
+	defer p.routesMu.Unlock()
+
+	var filteredRoutes []*RouteHandler
+	for _, route := range p.routes {
+		if route.path != path {
+			filteredRoutes = append(filteredRoutes, route)
+		}
+	}
+
+	p.routes = filteredRoutes
```

Comment:
Nit; I guess this could be simplified to:

```suggestion
p.routes = slices.DeleteFunc(p.routes, func(rh *RouteHandler) bool {
    return rh.path == path
})
```

PS: It would require adding the "slices" import.

Reply:
Good idea :+1: Updated in https://github.com/grafana/k6/pull/5223/commits/3337f162093506c1c212d7e8eeb9a998ee50e46b